source/extensions/network/dns_resolver/apple/apple_dns_impl.h

```diff
@@ -133,5 +134,19 @@ class AppleDnsResolverImpl : public DnsResolver, protected Logger::Loggable<Logg
   AppleDnsResolverStats stats_;
 };
 
+class AppleDnsResolverFactoryImpl : public DnsResolverFactory {
```

Comment:
You can move this class into the .cc file since it does not need to be public. Same for cares factory.

Reply:
ACK